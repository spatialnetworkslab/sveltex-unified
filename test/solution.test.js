
import dedent from 'dedent'
import { processor } from '../src/index.js'

const process = async (md) => {
  const result = await processor()
    .process(md)
  return result.toString()
}

test('process solution block', async () => {
  const input = await process(dedent`
          ::: solution show
            content
          :::
      `)
  expect(input).toMatchSnapshot()
})

test('process nested solution block', async () => {
  const input = await process(dedent`
::: solution show
    ::: codesandbox ./test/codesandbox/ some?props
    :::
:::
        `)
  expect(input).toMatchSnapshot()
})

test('use frontmatter to hide solution', async () => {
  const input = await process(dedent`
---
solution: false
---
::: solution
don't show this
:::
          `)
  expect(input).toMatchSnapshot()
})

test('local show option overrides frontmatter', async () => {
  const input = await process(dedent`
---
solution: false
---
::: solution show
    do show this
:::
            `)
  expect(input).toMatchSnapshot()
})
