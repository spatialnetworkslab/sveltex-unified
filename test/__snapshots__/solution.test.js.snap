// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local show option overrides frontmatter 1`] = `"<div class=\\"solution\\"><details><summary>Solution</summary><div class=\\"solutionContent solutionShow\\"><p>do show this</p></div></details></div>"`;

exports[`process nested solution block 1`] = `
"<div class=\\"solution\\"><details><summary>Solution</summary><div class=\\"solutionContent solutionShow\\"><iframe
  src=\\"https://codesandbox.io/embed/ozh6m?some?props\\"
  sandbox=\\"allow-modals allow-forms allow-popups allow-scripts allow-same-origin\\"
  style=\\"width:100%; height:500px; overflow:hidden;\\"
  class=\\"csb\\"
  title=\\"CodeSandbox for ./test/codesandbox/\\"
></iframe></div></details></div>"
`;

exports[`process solution block 1`] = `"<div class=\\"solution\\"><details><summary>Solution</summary><div class=\\"solutionContent solutionShow\\"><p>  content</p></div></details></div>"`;

exports[`use frontmatter to hide solution 1`] = `"<div class=\\"solution\\"><details><summary>Solution</summary></details></div>"`;
